config/confighttp/confighttp_test.go

```diff
@@ -420,6 +423,31 @@ func TestHttpReception(t *testing.T) {
 				assert.NoError(t, errRead)
 				assert.Equal(t, "test", string(body))
 			}
+
+			// Check that h2c payloads are handled when serving insecure traffic
+			if tt.tlsClientCreds.Insecure {
```

Comment:
Consider to have this as a separate test only for this config. It is not recommended to add this and be run only for one of the "sub-tests".

Reply:
@bogdandrutu done!